public/js/calendarPage.js

```diff
@@ -10,15 +11,26 @@ const calendar = new FullCalendar.Calendar(calendarEl, {
 });
 
 (async () => {
-  const response = await fetch("./api/getCalendarEvents");
+  const response = await fetch("/api/getAllCalendarData");
   const calendarEvents = await response.json();
-  calendarEvents.map((calendarEvent) => {
-    const categoryName = calendarEvent.categoryName
-      .toLowerCase()
-      .split(" ")
-      .join("");
-    calendarEvent.className = categoryName;
+
+  calendarEvents['eventData'].map((calendarEvent) => {
+    calendarEvent.className = calendarEvent.categoryName
+        .toLowerCase()
+        .split(" ")
+        .join("");
     calendar.addEvent(calendarEvent);
   });
+
+  calendarEvents['courseData'].map((calendarEvent) => {
+    calendarEvent.map((course) => {
+      course.className = course.categoryName
+        .toLowerCase()
+        .split(" ")
+        .join("");
+      calendar.addEvent(course)
+    })
+  });
```

Comment:
Since you're doing the same thing in `map` twice here, you could create a named function and use it instead to avoid repetition